### PR TITLE
tool: add Mergify

### LINKED ISSUE #4468: GitHub bot for auto-update/merge of PRs
We should consider setting up a GitHub bot like [Bulldozer](https://github.com/palantir/bulldozer) that can automatically update and/or merge pull requests once all required conditions are met - typically including labels like `auto-update` and `auto-merge`. It gets pretty tedious to manually wait for tests, update from master, wait again, and merge.

We'll need to get an admin to hook this up.

### PR DESCRIPTION
**Pull request by marbar3778**
## Description

- auto update and/or merge pull requests

- we currently wait for ci to pass then come back and merge it this will eliminate that step and auto merge if the specified labels are present

closes #4468

Signed-off-by: Marko Baricevic <marbar3778@yahoo.com>

______

For contributor use:

- [x] Wrote tests
- [x] Updated CHANGELOG_PENDING.md
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated relevant documentation (`docs/`) and code comments
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

**Comment by marbar3778** [comment]
this is not on the github marketplace need to find an alternative tool. will close for now

File: .mergify.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+pull_request_rules:
+  - name: automatic merge for master when CI passes and 1 reviews and has label S:merge
+    conditions:
+      - "#approved-reviews-by>=1"
+      - "status-success=ci/circleci: test-cover"
+      - base=master
+      - label=S:merge
```
Comment (erikgrinaker) [inline .mergify.yml]: Maybe `automerge`? Just to make it clear that it's not advisory, it actually causes things to happen.

Comment (alexanderbez) [reply .mergify.yml]: ACK `automerge`.

**Comment by marbar3778** [comment]
@tessr could you add the bot: https://github.com/marketplace/mergify it states that it is free for open source

**Pull request merged**